ti/instrumentation/utils.go

```diff
@@ -264,6 +267,151 @@ func getChangedFiles(ctx context.Context, workspace string, log *logrus.Logger,
 	return res, nil
 }
 
+// addBazelFilesToChangedFiles takes a list of files and removes bazel build files and adds java files listed in target src globs
+func addBazelFilesToChangedFiles(ctx context.Context, workspace string, log *logrus.Logger, res []ti.File, bazelFileCount string) ([]ti.File, []string, error) {
+	var moduleList []string
+	threshold, err := strconv.Atoi(bazelFileCount)
+	if err != nil {
+		return res, moduleList, fmt.Errorf("bazelFileCount not set correctly in ticonfig.yml, expecting number found character %v ", bazelFileCount)
+	}
+
+	// map to prevent duplicate files
+	uniqueFiles := make(map[string]struct{})
+
+	var changedRes []ti.File
+	for _, file := range res {
+		if !strings.HasSuffix(file.Name, "BUILD.bazel") {
+			// add non BUILD.bazel files to the changed files result directly and skip remainder
+			if _, exists := uniqueFiles[file.Name]; !exists {
+				changedRes = append(changedRes, file)
+				uniqueFiles[file.Name] = struct{}{}
+			}
+			continue
+		}
+		// If BUILD.Bazel at the root of harness-core is modified, run all tests /... by keeping BUILD.Bazel in changed files list
+		if strings.Replace(file.Name, "BUILD.bazel", "", -1) == "" {
+			log.Infoln("Changed file list is: ", res)
```

Comment:
The `res` is not changing in this loop but we are printing it every-time. Should be printed once.

Reply:
It would only be printed once for root BUILD.bazel file and we'll break out of the func.